### DISCUSSION TITLE
Fire TV HDMI button

### DISCUSSION THREAD

**Discussion by mbriacuni** [PRProd/HA-Firemote#235] [Q&A] [answered]
Hi,
I would need to know the adb command to change hdmi source.
I recently bought a Fire TV and the commands from the card work, but I would like to use them for automation and I'm not succeeding. I would also like to know if there is a command that switches between HDMI 1 and 2.
Thanks in advance

**Comment by PRProd** [comment, marked as answer]
Hello @mbriacuni!

YES!  I WILL ABSOLUTELY HELP YOU!  I remember being _very_ frustrated trying to figure this out for myself, only to find out that it was actually a pretty simple thing to do.
<br>
## Example 1: 
This script will switch my FireTV (media_player.fire_tv_192_168_107_88) to HDMI input 3
```yaml
service: androidtv.adb_command
data:
  command: HDMI3
target:
  entity_id: media_player.fire_tv_192_168_107_88
```
<br>

## Example 2:
You can also use the Developer tools to make this happen.

![image](https://github.com/PRProd/HA-Firemote/assets/25486599/b680700b-1ccf-4e80-ab15-c3322ac4bc09)
 1. Click on **Developer Tools**
 2. Click on **Services**
 3. Select **Android Debug Bridge: ADB command** as the service
 4. Choose your Fire TV entity as the target
 5. Type ```HDMI3``` into the command window
 6. Click the **CALL SERVICE** button

<br>
<br>

Choosing any HDMI input is that simple, just select one from this list and use that value as the command.  Please note: there are no spaces between the letters and the number.
 - HDMI1
 - HDMI2
 - HDMI3
 - HDMI4

I hope this helps!

**Reply by mizuxe** [reply]
This works for my Fire TV flawlessly, but not for my Google TV. Do you know if there is a separate command for Google TV?

**Comment by mbriacuni** [comment]
Thank you very much @PRProd! As far as you know, is there a list of commands somewhere? As I said before, I would also need the command to switch to the TV.

**Reply by PRProd** [reply]
@mbriacuni 

You're very welcome!  I'm happy to help!
<br>

> As far as you know, is there a list of commands somewhere?

There are commands scattered around the internet, but I haven't found one perfect source yet.  Right now, I think that the best source I have is my Firemote project itself!  Nearly everything I've researched across multiple devices has all been pulled together here.

I'm toying with the idea of adding a special feature to Firemote that will help answer questions like this, kind of like a built-in teacher for curious minds like yours... I'll let you know if I end up doing that or not.
<br>

> As I said before, I would also need the command to switch to the TV.

Can you elaborate a little bit here?  I don't quite understand what you are trying to figure out.

**Reply by mbriacuni** [reply]
Yes, sorry if I'm unclear, but I'm not very good at speaking English. What I wanted to know is if there is a command to select live tv from the antenna as the source. 

Surely, if you did something like that, you would be doing the whole community a great favor. You can hardly find anything clear around.

Thank you again for your help and the great job you have done.

**Reply by PRProd** [reply]
Are you asking for the command for this button?
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/1e727fc2-ba63-4924-b655-123fc7ecba42)

**Reply by PRProd** [reply]
@mbriacuni - I don't know if you found the answer on your own or not, but I've been working on making my documentation a lot more helpful.  I created a chart of [ADB commands for the Amazon Fire TV 4 Series](https://github.com/PRProd/HA-Firemote/wiki/Smart-TV---Fire-TV-4-Series-(2021)#adb-commands) which should include every answer you are looking for.

Here are the relevant entries that might answer your question:
| Action                    | ADB Command(s)                 | 
| ----------| ----------- |
| Antenna/Cable Input       | `adb shell input keyevent KEYCODE_TV`<br>`adb shell input keyevent 170` |     
| Antenna/Cable Switch      | `adb shell input keyevent KEYCODE_TV_ANTENNA_CABLE`<br>`adb shell input keyevent 242`       |
| Live TV         | `adb shell 'sendevent /dev/input/event0 1 362 1 && sendevent /dev/input/event0 0 0 0 && sendevent /dev/input/event0 1 362 0 && sendevent /dev/input/event0 0 0 0'`<br>`adb shell input keyevent 297`          |

**Reply by mbriacuni** [reply]
Oh, sorry I didn't answer anymore. Between work and other commitments, it has slipped my mind 🙇. In the meantime I replaced the fire tv so I didn't need to solve the problem anymore, but I got a monitor to which I connected a hybrid tv decoder with android tv and I'm sure the documentation you created will be very useful 👍
